app.py

```diff
@@ -0,0 +1,423 @@
+           isBackground:    True if the export of the asset happened in the background.
+           backgroundJobId: Id of the background job given by the backburner manager upon submission. 
+                            Empty if job is done in foreground.
+           width:           Frame width of the exported asset.
+           height:          Frame height of the exported asset.
+           aspectRatio:     Frame aspect ratio of the exported asset.
+           depth:           Frame depth of the exported asset. ( '8-bits', '10-bits', '12-bits', '16 fp' )
+           scanFormat:      Scan format of the exported asset. ( 'FILED_1', 'FIELD_2', 'PROGRESSIVE' )
+           fps:             Frame rate of exported asset.
+           sequenceFps:     Frame rate of the sequence the asset is part of.
+           sourceIn:        Source in point in frame and asset frame rate.
+           sourceOut:       Source out point in frame and asset frame rate.
+           recordIn:        Record in point in frame and sequence frame rate.
+           recordOut:       Record out point in frame and sequence frame rate.
+           track:           ID of the sequence's track that contains the asset.
+           trackName:       Name of the sequence's track that contains the asset.
+           segmentIndex:    Asset index (1 based) in the track.       
+           versionName:     Current version name of export (Empty if unversioned).
+           versionNumber:   Current version number of export (0 if unversioned).
+
+        :param comments: Review comments entered by the user.
+         
+        """ 
+        
+        # create version
+        full_path = os.path.join(info["destinationPath"], info["resolvedPath"])
+
+        if not os.path.exists(full_path):
+            raise TankError("Cannot find quicktime '%s'! Aborting upload." % full_path)
+        
+        self.log_debug("Begin shotgun processing for %s..." % full_path)
+        self.log_debug("File size is %s bytes." % os.path.getsize(full_path))
+                
+        # ensure that the entity exists in Shotgun
+        entity_name = info["sequenceName"]
+        entity_type = self.get_setting("shotgun_entity_type")
+
+        sg_data = self.shotgun.find_one(entity_type, [["code", "is", entity_name], 
+                                                      ["project", "is", self.context.project]]) 
+        
+        if not sg_data:    
+            # Create a new item in Shotgun
+            # First see if we should assign a task template
+            self.log_debug("Creating a new item in Shotgun...")
+            task_template_name = self.get_setting("task_template")
+            task_template = None
+            if task_template_name: 
+                task_template = self.shotgun.find_one("TaskTemplate", [["code", "is", task_template_name]])
+                if not task_template:
+                    raise TankError("The task template '%s' specified in the task_template setting "
```

Comment:
Reporting it only once as it seems to be a common pattern : I don't feel like it's right to have a "!" at the end of an error message.